src/components/UserListScreen.js

```diff
@@ -16,7 +46,21 @@ const UserListScreen = ({
   navigation,
   route,
 }: UserListScreenProps): React.Node => {
-  const users = useSelector((state: UserState) => state.users);
+  const users = useSelector((state: RootState) => state.user.users);
+  const dispatch = useDispatch<UserDispatch>();
+  const isFetching = useSelector<RootState, boolean>(isFetchingUsersSelector);
+  const fetchedUsers = useSelector<RootState, User[]>(fetchUsersSelector);
+  const fetchUsersError = useSelector<RootState, any>(fetchUsersErrorSelector);
+
+  React.useEffect(() => {
+    dispatch(fetchUsersAC());
+  }, [dispatch]);
+
+  React.useEffect(() => {
+    if (!isFetching && !fetchUsersError) {
+      dispatch(addUsers(fetchedUsers));
+    }
+  }, [dispatch, fetchedUsers, isFetching, fetchUsersError]);
```

Comment:
addUsers(fetchedUsers) logic should not place here. We should use middleware to catch redux-api-call actions and addUsers in middleware too

Reply:
Thank you for your feedback. I've used redux-observable to addUsers instead of explicitly dispatching an action here